server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -174,38 +174,21 @@ def _filter_annotation(self, filter, index, axis):
         return index
 
     @cache.memoize()
-    def metadata_ranges(self, df=None):
-        metadata_ranges = {}
-        if not df:
-            df = self.data
-        for field in self.schema:
-            if self.schema[field]["variabletype"] == "categorical":
-                group_by = field
-                if group_by == "CellName":
-                    group_by = "cell_name"
-                metadata_ranges[field] = {"options": df.obs.groupby(group_by).size().to_dict()}
-            else:
-                metadata_ranges[field] = {
-                    "range": {
-                        "min": df.obs[field].min(),
-                        "max": df.obs[field].max()
-                    }
-                }
-        return metadata_ranges
-
-    @cache.memoize()
-    def metadata(self, df, fields=None):
+    def annotation(self, df, fields=None):
```

Comment:
does anyone use the fields param?

Reply:
yep, that's what is coming from the annotation-names filter